scipy/spatial/tests/test_kdtree.py

```diff
@@ -1254,16 +1254,15 @@ def test_kdtree_duplicated_inputs(kdtree_type):
         # it shall not divide more than 3 nodes.
         # root left (1), and right (2)
         kdtree = kdtree_type(data, leafsize=1)
```

Comment:
What about adding a query test here? Is the new balanced tree build covered by a test?

Reply:
I've fixed the unbalanced tree to use strict less-than, which means this tree always generates exactly 3 leaves. So, I can just assert that all the indices from the ones are in the first leaf and all the twoes in the second.